spec/index.bs

```diff
@@ -8,16 +8,1400 @@ URL: https://wicg.github.io/trusted-types/dist/spec/index.html
+specified in the violation’s {{TrustedTypeConfiguration|configuration
+object}}.
+
+1.  Enqueue a task to run the following steps:
+    1.  Let *group* be *violation*’s configuration's
+        {{TrustedTypeConfiguration/reportingGroup}}.
+    1.  Let *settings* be *violation*’s [[CSP3#violation-global-object|global object]]’s
+        relevant settings object.
+    1.  Execute [[REPORTING#queue-report|Queue data as type for endpoint group on settings algorith]]
+        with the following arguments:
+        *  *data* = violation
+        *  *type* = `“trusted-types”`
+        *  *endpoint group* = *group*
+        *  *settings* = *settings*
+
+        Note: The optional *url* parameter is not provided.
+
+# Integrations # {#integrations}
+
+Note: DOM sink setters may accept these types to indicate that they accept
+the appropriate trusted type or a string when enforcement is off or when
+enforcement falls back to a [=default policy=].
+
+<xmp class="idl">
+typedef (DOMString or TrustedHTML) HTMLString;
+typedef (DOMString or TrustedScript) ScriptString;
+typedef (USVString or TrustedScriptURL) ScriptURLString;
+typedef (USVString or TrustedURL) URLString;
+typedef (TrustedHTML or TrustedScript or TrustedScriptURL or TrustedURL) TrustedType;
+</xmp>
+
+## Algorithms ## {#integrations-algorithms}
+
+### <dfn abstract-op>Get Trusted Type compliant string algorithm</dfn> ### {#get-trusted-type-compliant-string-algorithm}
+
+This algorithm will return a string that can be assigned to a DOM
+[=sink=], optionally unwrapping it from a matching [=Trusted Type=].
+It will ensure that the Trusted Type enforcement setting was respected.
+
+Given a TrustedType type (*expectedType*), a {{Document}}
+(*document*), a {TrustedType} (*input*), and a boolean
+(*passThroughFunctions*) run these steps:
+
+ISSUE: *input* could also be a {{Function}}, e.g. for `setTimeout`.  Maybe add `or Function` to `ScriptString` and rename to avoid confusion.
+
+ISSUE: Make it clear that no reporting is done when *disposition* is `unsafe-allow`.
+
+ISSUE: In step 1, refer to the internal slot in Document or the IDL additions to document that store the configuration.
+
+1.  Let *disposition* be *document*’s TrustedTypeConfiguration's
+    {{TrustedTypeConfiguration/domSinks}}.
+1.  If *disposition* is `“unsafe-allow”`, then run these steps:
+    1.  If *passThroughFunctions* is true and *input* is a {{Function}},
+        return *input* and abort these steps.
+    1.  Otherwise, return stringified *input* and abort these steps.
+1.  If *input* has type *expectedType*, return stringified
+    *input* and abort these steps.
+1.  If *passThroughFunctions* is true and *input* is a {{Function}},
+    return input and abort these steps.
+1.  Otherwise, execute the following steps:
+    1.  Let *defaultPolicy* be the result of executing
+        [$Get default policy algorithm$] on *document*’s
+        [[HTML5#concept-document-bc|browsing context]]'s {{Window}}
```

Comment:
What did you mean by `#concept-document-bc`? It's not in the HTML5 spec or in the Google Docs doc.

Reply:
Should this be https://html.spec.whatwg.org/multipage/browsers.html#browsing-context, @mikesamuel  ?

Reply:
No.  That define's the type browsing context, but not the algorithm for finding the one associated with the current document.

https://html.spec.whatwg.org/multipage/browsers.html#concept-document-bc is what I was trying to link to.

> A Document's browsing context is the browsing context whose session history contains the Document, if any such browsing context exists and has not been discarded, and null otherwise.

https://html.spec.whatwg.org/multipage/window-object.html#window-bc might work too.

> The Window object's browsing context is the Window object's associated Document's browsing context. It is either null or a browsing context.